### DISCUSSION TITLE
Limitation on the allowed number of states per site

### DISCUSSION THREAD

**Discussion by mathew0036** [QuSpin/QuSpin#628] [Q&A]
I am working with a custom Hilbert space and I wanted construct a unit vector corresponding to one of the basis vectors. I followed by first defining the index corresponding to the string representation of the basis vector using basis.index(). But I got an error " **ValueError: int() base must be >= 2 and <=36, or 0** ". This seems to come from the int() function inside the basis_general.py file. Is there a work around for this problem?

**Comment by mgbukov** [comment]
Did you go thru the basis tutorial? I think it should help:

http://quspin.github.io/QuSpin/examples/example00.html#example-00-label

**Reply by mathew0036** [reply]
That is precisely what I did. My states per site are quite large, as I deal with a bosonic+fermionic system. 
I used the construction from line 80 to line 86, and the int() function, which converts the string to an integer, is limited to only base 36, while my states per site can go well beyond that.

**Comment by mgbukov** [comment]
Ah, I see; this looks like a large integer support problem.

Unfortunately, at the moment, you'd have to find another way to do this.
E.g., you can open up the python code in quspin to see how this is
implemented and see if there's a way to do something similar with your
basis w/o using the int function.


On Wed, May 31, 2023, 10:32 AM mathew0036 ***@***.***> wrote:

> That is precisely what I did. My states per site are quite large, as I
> deal with a bosonic+fermionic system.
> I used the construction from line 80 to line 86, and the int() function,
> which converts the string to an integer, is limited to only base 36, while
> my states per site can go well beyond that.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QuSpin/QuSpin/discussions/628#discussioncomment-6047277>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHSUBSHTTN5QUJ7DNZ5763XI36ZXANCNFSM6AAAAAAYVBH7VA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
Marin
